src/lib.rs

```diff
@@ -1,59 +1,94 @@
-extern crate errno;
+#![deny(warnings)]
+#![deny(missing_docs)]
+
+//! # rust-zonename
+//!
+//! The `rust-zonename` crate provides a rust abstraction over the illumos `getzoneid`,
+//! `getzoneidbyname`, and `getzonenamebyid` C library functions.
+
 extern crate libc;
 
+use libc::{c_char, size_t};
 use std::ffi::CString;
-use libc::{c_char,size_t};
-use errno::{Errno,errno};
+use std::io;
 
 mod ffi;
 
+/// Max length of a zone name.
 pub const ZONENAME_MAX: usize = 64;
 
-pub fn getzoneid() -> Result<i32, Errno> {
-    let zoneid = unsafe {
-        ffi::getzoneid()
-    };
+/// Returns the zone ID of the calling process.
+///
+/// # Example
+/// ```
+/// let zoneid = zonename::getzoneid().expect("failed to get zoneid");
+/// ```
+pub fn getzoneid() -> io::Result<i32> {
```

Comment:
You don't have to specify the error type when returning an `io::Result`? this is really convenient

Reply:
Yeah its basically a typedef:
https://doc.rust-lang.org/std/io/type.Result.html